aiohttp/web_request.py

```diff
@@ -151,15 +151,30 @@ def secure(self):
         return self.url.scheme == 'https'
 
     @reify
+    def _forwarded(self):
+        forwarded = self._message.headers.get(hdrs.FORWARDED)
+        if forwarded is not None:
+            parsed = re.findall(
+                r'^by=([^;]*); +for=([^;]*); +host=([^;]*); +proto=(https?)$',
```

Comment:
I suspect the parser should be more complicated.
```
The header field value can be
   defined in ABNF syntax as:

       Forwarded   = 1#forwarded-element

       forwarded-element =
           [ forwarded-pair ] *( ";" [ forwarded-pair ] )

       forwarded-pair = token "=" value
       value          = token / quoted-string

       token = <Defined in [RFC7230], Section 3.2.6>
       quoted-string = <Defined in [RFC7230], Section 3.2.6>
```
It means the `forwarded-pair` elements could be in random order: `by=a; host=b.com` and `host=b.com, by=a` are both valid combinations.
Also value could be optionally quoted, aiohttp shoud unquote it automatically. 
pair-name is case-insensitive: all 'Host', 'host', and 'HOst` are valid names:
```
       Forwarded: for="_gazonk"
       Forwarded: For="[2001:db8:cafe::17]:4711"
       Forwarded: for=192.0.2.60;proto=http;by=203.0.113.43
       Forwarded: for=192.0.2.43, for=198.51.100.17

   Note that as ":" and "[]" are not valid characters in "token", IPv6
   addresses are written as "quoted-string".
```